tests/cli/test_datasets.py

```diff
@@ -1316,3 +1329,23 @@ def test_add_removes_local_path_information(runner, client, directory_tree):
         for file_ in dataset.files:
             assert file_.url.startswith('file://../')
             assert file_.url.endswith(file_.name)
+
+
+def test_add_remove_credentials(runner, client, monkeypatch):
+    """Check removal of credentials during adding of remote data files."""
+    url = 'https://username:password@example.com/index.html'
+
+    def get(u):
+        """Mocked response."""
+        response = requests.Response()
+        response._content = b'{}'
+        return response
+
+    result = runner.invoke(cli, ['dataset', 'create', 'my-dataset'])
+    assert 0 == result.exit_code
+
+    monkeypatch.setattr(requests, 'get', get)
+    dataset = client.load_dataset('my-dataset')
+    o = client._add_from_url(dataset, url, client.path)
```

Comment:
Is there a way to avoid calling a non-public method?

Reply:
That client abstraction should really be fixed (IMHO), cause isolating pieces in testable units is almost impossible without resorting to such calls.

Reply:
I don't get why you removed the `cli` call to add files. I thought perhaps it was because of the requests mocking that something wasn't working in between.
Perhaps it's better to remove this test and have it only as integration test because it is mocking an implementation detail and it is calling a protected method.

Reply:
I din't removed anything, that test the way it was written it was integration test and it was not marked as such - I just marked it as an integration test and wrote the actual unit test for the part of what that test is suppose to be testing since @rokroskar requested to have it.

I would also be fine just with having it as an integration test, tbh.